nerfstudio/data/pixel_samplers.py

```diff
@@ -300,3 +343,59 @@ def sample_method(
             indices = indices.flatten(0, 2)
 
         return indices
+
+
+@dataclass
+class PairPixelSamplerConfig(PixelSamplerConfig):
+    """Config dataclass for PatchPixelSampler."""
+
+    _target: Type = field(default_factory=lambda: PairPixelSampler)
+    """Target class to instantiate."""
+    radius: int = 2
+    """max distance between pairs of pixels."""
+
+
+class PairPixelSampler(PixelSampler):  # pylint: disable=too-few-public-methods
+    """Samples pair of pixels from 'image_batch's. Samples pairs of pixels from
+        from the images randomly within a 'radius' distance apart. Useful for pair-based losses.
+
+    Args:
+        config: the PairPixelSamplerConfig used to instantiate class
+    """
+
+    def __init__(self, config: PairPixelSamplerConfig, **kwargs) -> None:
+        self.config = config
+        self.radius = self.config.radius
+        self.rays_to_sample = self.config.num_rays_per_batch // 2
+        super().__init__(self.config, **kwargs)
+
+    # overrides base method
+    def sample_method(  # pylint: disable=no-self-use
+        self,
+        batch_size: int,
+        num_images: int,
+        image_height: int,
+        image_width: int,
+        mask: Optional[Tensor] = None,
+        device: Union[torch.device, str] = "cpu",
+    ) -> Int[Tensor, "batch_size 3"]:
+        if mask:
```

Comment:
I think this should be: `if mask is None`. My `transforms.json` contains masks and as a result the code fails when `maks` tensor does not evaluate as a boolean expression.

Maybe it would be possible to add instructions to the `NotImplementedError` message on what needs to be adjusted in the configuration for the code to work with mask images (my case)? Naybe automatically fall back to the PixelSampler, if that's the solution?